File: app/src/main/java/forpdateam/ru/forpda/presentation/auth/AuthView.java
package forpdateam.ru.forpda.presentation.auth;

import forpdateam.ru.forpda.api.auth.models.AuthForm;
import forpdateam.ru.forpda.api.profile.models.ProfileModel;
import forpdateam.ru.forpda.common.mvp.IBaseView;

/**
 * Created by radiationx on 02.01.18.
 */

public interface AuthView extends IBaseView {

    void showForm(AuthForm authForm);

    void showLoginResult(boolean success);

    void showProfile(ProfileModel profile);
}

File: app/src/main/java/forpdateam/ru/forpda/presentation/auth/AuthView.kt
package forpdateam.ru.forpda.presentation.auth

import forpdateam.ru.forpda.entity.remote.auth.AuthForm
import forpdateam.ru.forpda.entity.remote.profile.ProfileModel
import moxy.MvpView
import moxy.viewstate.strategy.AddToEndSingleStrategy
import moxy.viewstate.strategy.StateStrategyType

/**
 * Created by radiationx on 02.01.18.
 */

@StateStrategyType(AddToEndSingleStrategy::class)
interface AuthView : MvpView {
    fun setSendEnabled(isEnabled: Boolean)
    fun setSendRefreshing(isRefreshing: Boolean)
    fun onFormLoaded(authForm: AuthForm)
    fun onSuccessAuth()
    fun showProfile(profile: ProfileModel)
}
